enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+as `capabilities` or not)
+* Providing a way to install OLM operators as part of the initial cluster install.  This EP is about making
+the install experience around the existing CVO-based components more flexible, not adding new components to the
+install experience.
+* Allowing components to be disabled post-install.
+* Eliminating or replacing cluster profiles
+* Encoding logic in the installer itself about which components should be disabled under specific circumstances
+
+
+## Proposal
+
+### User Stories
+
+* As a user creating a new cluster that will be managed programmatically, I do not want the additional
+security exposure and resource overhead of running the web console.  I would like a way to install
+a cluster that has no console out of the box, rather than having to disable it post-install or
+modify rendered manifests in a way that requires deep understanding of the OCP components/resources.
+
+* As a team scaffolding a managed service based on openshift, I want to minimize the footprint of my
+clusters to the components I need for the service.
+
+* As a user creating a cluster that will never run an image registry, I do not want the additional overhead
+of running the image registry operator, or have to remove the default registry that is created.
+
+* As a team using openshift for a specific use case such as edge deployments, I want to provide
+an install experience that disables components that aren't needed for my use case.
+
+### API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The CVO already has the ability to respect annotations on resources, as can be seen
+[here](https://github.com/openshift/cluster-kube-apiserver-operator/blob/c03c9edf5fddf4e3fb1bc6d7afcd2a2284ca03d8/manifests/0000_20_kube-apiserver-operator_06_deployment.yaml#L10) and leveraged [here](https://github.com/openshift/hypershift/blob/b98bc43467a73921891ede7d6477757b647f1985/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L152-L153).
+This proposal consists of two parts:
+
+1) Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
+
+```yaml
+annotations:
+  capability.openshift.io: "console"
+```
+
+The `oc adm release new` command will be updated to extract these annotations and embed a list of the valid
+capabilities in the release payload image.  This list can be consulted by users to determine the set of
+capabilities they can enable/disable when configuring the installation.
+
+Note: Resources are enabled unless explicitly excluded by configuration.  In the future we may allow
+resources to be excluded unless explicitly included, this can be done by introducing a new annotation
+such as `capability.openshift.io/defaultDisable`.  If we do so, we will need to define how that choice
```

Comment:
Nit: but why not `capability.openshift.io/default` with value being enabled, disabled, or something else if we need to. The current assumes only boolean values.

Reply:
`default` isn't really clear as to what it is defaulting, so i'd probably want a more descriptive name that makes it clear it represents the "value for whether or not this capability is enabled by default", but something that allows for more values than true+false makes sense.  We can have that discussion in more depth if/when we actually go to introduce this as a supported annotation.

Reply:
my thinking was this `capability.openshift.io` has this `default` and the value says what, but like I said, it's a nit that we can figure out in the implementation phase when more things will be on the table :wink: